panther_hardware_interfaces/src/panther_system/robot_driver/roboteq_robot_driver.cpp

```diff
@@ -0,0 +1,225 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "panther_hardware_interfaces/panther_system/robot_driver/robot_driver.hpp"
+
+#include <chrono>
+#include <ctime>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <thread>
+#include <utility>
+
+#include "lely/util/chrono.hpp"
+
+#include "panther_hardware_interfaces/panther_system/robot_driver/canopen_manager.hpp"
+#include "panther_hardware_interfaces/panther_system/robot_driver/roboteq_data_converters.hpp"
+#include "panther_hardware_interfaces/panther_system/robot_driver/roboteq_driver.hpp"
+#include "panther_hardware_interfaces/panther_system/robot_driver/roboteq_robot_driver.hpp"
+
+namespace panther_hardware_interfaces
+{
+
+RoboteqRobotDriver::RoboteqRobotDriver(
+  const CANopenSettings & canopen_settings, const DrivetrainSettings & drivetrain_settings,
+  const std::chrono::milliseconds activate_wait_time)
+: canopen_settings_(canopen_settings),
+  drivetrain_settings_(drivetrain_settings),
+  canopen_manager_(canopen_settings),
+  roboteq_vel_cmd_converter_(drivetrain_settings),
+  pdo_motor_states_timeout_ms_(canopen_settings.pdo_motor_states_timeout_ms),
+  pdo_driver_state_timeout_ms_(canopen_settings.pdo_driver_state_timeout_ms),
+  activate_wait_time_(activate_wait_time)
+{
+}
+
+void RoboteqRobotDriver::Initialize()
+{
+  if (initialized_) {
+    return;
+  }
+
+  try {
+    canopen_manager_.Initialize();
+    DefineDrivers();
+    for (auto & [name, driver] : drivers_) {
+      data_.emplace(name, RoboteqData(drivetrain_settings_));
+      driver->Boot();
+    }
+  } catch (const std::runtime_error & e) {
```

Comment:
You throw an exception with the same type once again. What is the point to catch it?